### PR TITLE
Added models for deep vs shallow copy object creation along with unit test

### PR DESCRIPTION
**Pull request by balasr21**
This PR contains code examples for creating deep vs shallow copy of an object

### CODE REVIEW

File: core-java-modules/core-java/src/main/java/com/baeldung/clone/DeepMatch.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+public class DeepMatch extends ShallowMatch {
+
+    public DeepMatch(TYPE type, OffsetDateTime startTime, OffsetDateTime endTime, Integer durationInMinutes, PlayersDetail playersDetail) {
+        super(type, startTime, endTime, durationInMinutes, playersDetail);
+    }
+
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
+
```
Comment (andrzejsydor) [inline core-java-modules/core-java/src/main/java/com/baeldung/clone/DeepMatch.java:13]: empty line could be deleted

File: core-java-modules/core-java/src/main/java/com/baeldung/clone/ShallowMatch.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    private OffsetDateTime startTime;
+
+    private OffsetDateTime endTime;
+
+    private Integer durationInMinutes;
+
+    private PlayersDetail playersDetail;
+
+    public enum TYPE {
```
Comment (andrzejsydor) [inline core-java-modules/core-java/src/main/java/com/baeldung/clone/ShallowMatch.java:22]: why not just `Type` (CamelCase)
It's the standard, I think, see: https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html

**Review by andrzejsydor — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java/src/main/java/com/baeldung/clone/DeepMatch.java (hunk 1 above)
Comment (balasr21) [reply core-java-modules/core-java/src/main/java/com/baeldung/clone/DeepMatch.java]: Thanks for your comments, done

File: core-java-modules/core-java/src/main/java/com/baeldung/clone/ShallowMatch.java (hunk 2 above)
Comment (balasr21) [reply core-java-modules/core-java/src/main/java/com/baeldung/clone/ShallowMatch.java]: Thanks for your comments, done

**Pull request closed without merging**